### ISSUE TITLE
Panic when printing transaction

### ISSUE THREAD

**Issue by CodeSandwich** [foundry-rs/foundry#8635]
### Component

Cast

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (7f0f5b4 2024-08-08T00:19:27.650028841Z)

### What command(s) is the bug in?

cast tx

### Operating System

Linux

### Describe the bug

I got this error when running `cast tx`:
```
The application panicked (crashed).
Message:  called `Option::unwrap()` on a `None` value
Location: crates/common/fmt/src/ui.rs:381

This is a bug. Consider reporting it at https://github.com/foundry-rs/foundry

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
   1: __libc_start_main<unknown>
      at <unknown source file>:<unknown line>

Run with COLORBT_SHOW_HIDDEN=1 environment variable to disable frame filtering.
Run with RUST_BACKTRACE=full to include source snippets.
Aborted (core dumped)
```
This happened when fetching a transaction from a Filecoin RPC, so the root cause may be stemming from an unexpected behavior on the RPC side, but it still shouldn't cause an uncaught panic. The code pointed at by the error (`<Transaction as UIfmt>::pretty`) is packed with `unwrap`s causing panics when any piece of information to be printed for whatever reason is unavailable. It would be great to provide some placeholders or skip the unavailable fields, in this case any information is better than an error message.

**Comment by mattsse** [comment]
could you perhaps post an example transaction hash?

**Comment by CodeSandwich** [comment]
Sure! The transaction hash is [0x8da6741ce3104199ed066a40928332e0ea86b4eebd36463fb2cb469996ccb225](https://filscan.io/en/message/0x8da6741ce3104199ed066a40928332e0ea86b4eebd36463fb2cb469996ccb225/), to reproduce the error run:
```
cast tx -r https://rpc.ankr.com/filecoin 0x8da6741ce3104199ed066a40928332e0ea86b4eebd36463fb2cb469996ccb225
```

**Comment by mattsse** [comment]
thanks, fixed in #8636

**Comment by mattsse** [comment]
until released you can use `cast rpc eth_getTransactionByHash "0x8da6741ce3104199ed066a40928332e0ea86b4eebd36463fb2cb469996ccb225" -r https://rpc.ankr.com/filecoin | jq`
